crates/trie/sparse/src/trie.rs

```diff
@@ -679,13 +724,18 @@ impl RevealedSparseTrie {
                 }
                 SparseNode::Branch { state_mask, hash } => {
                     if let Some(hash) = hash.filter(|_| !prefix_set_contains(&path)) {
-                        buffers.rlp_node_stack.push((path, RlpNode::word_rlp(&hash)));
+                        buffers.rlp_node_stack.push((
+                            path,
+                            RlpNode::word_rlp(&hash),
+                            false,
+                            SparseNodeType::Branch { store_in_db_trie: true },
```

Comment:
why is the flag set to true in this case?

Reply:
thanks, fixed by memoizing in the same way as we do with `hash`